src/main/java/duke/commands/ByeCommand.java

```diff
@@ -0,0 +1,26 @@
+package duke.commands;
+
+import duke.exceptions.DukeException;
+import duke.utils.Storage;
+import duke.utils.TaskList;
+import duke.utils.Ui;
+
+/** Command invoked when "bye" command is encountered. */
+public class ByeCommand extends Command {
+    /**
+     * Upon execution, will change the <code>isExit</code> boolean is true. Will also
+     * save the TaskList a final time before Duke exits.
+     *
+     * @param ui       Ui object that is responsible for printing output as a response
+     * @param storage  Storage object respnsible for saving the Tasks into a pre-defined format
+     * @param allTasks TaskList object containing all tasks.
+     * @throws DukeException re-thrown from underlying method calls.
+     */
+    public void execute(Ui ui, Storage storage, TaskList allTasks) throws DukeException {
+        //Will cause while loop in Duke to exit
+        this.isExit = true;
+
+        //Parent method invoked to save TaskList
+        super.execute(ui, storage, allTasks);
+    }
```

Comment:
Curious to know if the ByeCommand is able to execute and exit the program if a file path is not specified or broken. I encountered it on my code and yours look similar.

Reply:
Hmm I didn't encounter this issue in my code. For me, the file path is saved as a static variable, so it does not change once the application starts. Could you clarify on the issue you are facing? Do you mean to say that pass your ByeCommand object a file path?

Reply:
It's more of an edge case test, where the file path does not exist or is broken when the application starts. So when you try to save, a DukeException gets thrown and you aren't able to exit the program with a ByeCommand because the boolean variable was unable to set to True.